util/util.go

```diff
@@ -39,8 +40,10 @@ var (
 		"index.docker.io": "library",
 		"docker.io":       "library",
 	}
+
 	// Transports contains the possible transports used for images
 	Transports = map[string]string{
+		docker.Transport.Name():        "",
```

Comment:
Surprisingly, this probably does not work: the `ResolveName` use below returns the after-`:` part as a docker-like reference string, i.e. `docker://busybox` would return `//busybox`.  (AFAICT the way `ResolveName` and `resolveImage` work together to handle `docker://busybox` is _pretty_ non-obvious.)

Reply:
… ah, it maybe actually does, because `resolveImage` stitches `BuilderOptions.Transport` and `//busybox` back together.  Mostly, the `hasRegistry` at the end of `pullImage` will choke on it again.

_Please_ let’s not add yet another possible syntax of strings returned from `ResolveName` like that.